library/HTMLPurifier/AttrDef/CSS/Gradient.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+/**
+ * Validates Gradient as defined by CSS.
+ */
+class HTMLPurifier_AttrDef_CSS_Gradient extends HTMLPurifier_AttrDef
+{
+
+    /**
+     * @type HTMLPurifier_AttrDef_CSS_Color
+     */
+    protected $color;
+
+    /**
+     * @type HTMLPurifier_AttrDef_CSS_Direction
+     */
+    protected $direction;
+
+    /**
+     * @type HTMLPurifier_AttrDef_CSS_Angle
+     */
+    protected $angle;
+
+    /**
+     * @type HTMLPurifier_AttrDef_CSS_Composite
+     */
+    protected $size;
+
+    public function __construct()
+    {
+        $this->color = new HTMLPurifier_AttrDef_CSS_Color();
+        $this->direction = new HTMLPurifier_AttrDef_CSS_Direction();
+        $this->angle = new HTMLPurifier_AttrDef_CSS_Angle();
+        $this->size = new HTMLPurifier_AttrDef_CSS_Composite(
+            array(
+                new HTMLPurifier_AttrDef_CSS_Length(),
+                new HTMLPurifier_AttrDef_CSS_Percentage()
+            )
+        );
+    }
+
+    /**
+     * @param string $string
+     * @param \HTMLPurifier_Config $config
+     * @param \HTMLPurifier_Context $context
+     * @return bool|string
+     */
+    public function validate($string, $config, $context)
+    {
+        if (strpos($string, 'gradient(') === false) {
+            return false;
+        }
+
+        $gradientFunctions = array(
+            'repeating-linear-gradient',
+            'repeating-radial-gradient',
+            'linear-gradient',
+            'radial-gradient',
+        );
+
+        // Get gradient function (linear, radial, repeating-linear, repeating-radial)
+        $function = null;
+        foreach ($gradientFunctions as $gradientFunction) {
+            // If function exists and is in first position
+            if (strpos($string, $gradientFunction) === 0) {
+                $function = $gradientFunction;
+                break;
+            }
+        }
+
+        if (!$function) {
+            return false;
+        }
+
+        // Check if function is linear or radial
+        $linear = false;
+        if (strpos($function, 'linear') !== false) {
+            $linear = true;
+        }
+
+        preg_match('#gradient\((.*)\)#', $string, $values);
```

Comment:
This regex will get bamboozled if I have something like `repeating-linear-gradient-gradient()`, won't it?